### DISCUSSION TITLE
How to get the COUNT of all results in a resultset?

### DISCUSSION THREAD

**Discussion by natgross** [cofu-app/cbl-dart#496] [Q&A] [answered]
We have a count function which returns the count of ALL documents. Ok.
We can of course get the size/count of the current resultset. Ok.
But can I get a count of all documents [to be] returned from a query after all paging is complete?
Using limit and skip I page thru the query. But I want to know the complete count as well. I would like to show the user a message like "Now Showing X documents of a total of Y".
Thanks

**Comment by blaugold** [comment]
There is no specific API for that. The typical approach is to use a count query that only selects `count(*)` and does not have `LIMIT` and `SKIP`, but is otherwise identical to the query that is used to load the actual data.

**Reply by natgross** [reply]
Does count(*) incur an overhead like pulling in the _id for each document? If so, it still impacts ram significantly for HUGE resultsets, like millions of docs. (On the other hand if it only counts, the ram use would still be bearable.)

**Reply by blaugold** [reply, marked as answer]
I don't think count(*) counts unique results in any way, just the number of results.
